mlflow/sagemaker/__init__.py

```diff
@@ -316,36 +388,121 @@ def _upload_s3(local_model_path, bucket, prefix):
             return '{}/{}/{}'.format(s3.meta.endpoint_url, bucket, key)
 
 
-def _deploy(role, image_url, app_name, model_s3_path, run_id, region_name):
+def _deploy(role, image_url, app_name, model_s3_path, run_id, region_name, mode, archive):
     """
     Deploy model on sagemaker.
     :param role: SageMaker execution ARN role
     :param image_url: URL of the ECR-hosted docker image the model is being deployed into
     :param app_name: Name of the deployed app
     :param model_s3_path: s3 path where we stored the model artifacts
     :param run_id: RunId that generated this model
+    :param mode: The mode in which to deploy the application.
+    :param archive: If True, any pre-existing SageMaker application resources that become inactive
+                    (i.e. as a result of deploying in mlflow.sagemaker.DEPLOYMENT_MODE_REPLACE mode) 
+                    will be preserved. If False, these resources will be deleted.
     """
     sage_client = boto3.client('sagemaker', region_name=region_name)
-    model_name = app_name + '-model'
+    s3_client = boto3.client('s3', region_name=region_name)
+
+    deployed_endpoints_page = sage_client.list_endpoints(MaxResults=100)
+    endpoint_found = (app_name in [endp["EndpointName"] for endp in deployed_endpoints_page["Endpoints"]])
+    while (not endpoint_found) and ("NextToken" in deployed_endpoints_page):
+        next_token = deployed_endpoints_page["NextToken"]
+        deployed_endpoints_page = sage_client.list_endpoints(MaxResults=100, NextToken=next_token)
+        for endp in deployed_endpoints_page["Endpoints"]:
```

Comment:
could do 
```
endpoint_found = any((ep["EndpointName"] == app_name for ep in deployed_endpoints_page["Endpoints"]))
```

Reply:
Replaced with this nice one-liner. Thanks!